packages/nexus-sdk/src/File/index.ts

```diff
@@ -1,22 +1,192 @@
-import { toPromise } from '@bbp/nexus-link';
+import FormData from 'isomorphic-form-data';
+import { toPromise, FetchAs } from '@bbp/nexus-link';
+import {
+  GetFileOptions,
+  NexusFile,
+  FilePayload,
+  LinkFilePayload,
+  UpdateFilePayload,
+} from './types';
+import { buildQueryParams, isBrowser } from '../utils';
+import {
+  PaginatedResource,
+  ResourceListOptions,
+  TagResourcePayload,
+  GetResourceOptions,
+} from '../Resource/types';
+import { Observable } from '../../../nexus-link/lib';
 
-const NexusFile = ({ httpGet }, context: any) => {
+const NexusFile = (
+  { httpGet, httpPost, httpPut, httpDelete, poll },
+  context: any,
+) => {
   return {
     get: (
       orgLabel: string,
       projectLabel: string,
       fileId: string,
-    ): Promise<Blob> =>
-      toPromise(
+      options?: GetFileOptions,
+    ): Promise<NexusFile | Blob | string | FormData> => {
+      const { as = FetchAs.BLOB, ...opts } = options;
+      return toPromise(
         httpGet({
           path: `${context.uri}/${
             context.version
-          }/files/${orgLabel}/${projectLabel}/${fileId}`,
+          }/files/${orgLabel}/${projectLabel}/${fileId}${buildQueryParams(
+            opts,
+          )}`,
           context: {
-            as: 'blob',
+            as,
           },
         }),
+      );
+    },
+
+    list: (
+      orgLabel: string,
+      projectLabel: string,
+      options?: ResourceListOptions,
+    ): Promise<PaginatedResource<NexusFile>> => {
+      const opts = buildQueryParams(options);
+      return toPromise(
+        httpGet({
+          path: `${context.uri}/${
+            context.version
+          }/files/${orgLabel}/${projectLabel}${opts}`,
+        }),
+      );
+    },
+
+    create: (
+      orgLabel: string,
+      projectLabel: string,
+      payload: FilePayload,
+    ): Promise<NexusFile> => {
+      const { '@id': fileId, file, storage } = payload;
+      const body = new FormData();
+      body.append('file', file);
+      // if in Node.js, we need to manually set headers
+      const headers = isBrowser ? {} : body.getHeaders();
```

Comment:
and maybe keep this 
```suggestion
      const headers = isBrowser ? {} : file.getHeaders();
```
of add an extra option field, like `extraHeaders`...